src/components/common/Dialog/dialogs/RejestracjaEmail/RejestracjaEmail.js

```diff
@@ -0,0 +1,139 @@
+import { makeStyles } from '@material-ui/core';
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import { openDialog, FormType } from "../../store/dialogSlice";
+import { theme } from '../../../../../theme/theme';
+import { ThemeProvider } from '@material-ui/core/styles';
+
+const useStyles = makeStyles({
+  field: {
+    marginTop: 15,
+    marginBottom: 20,
+  }
+})
+
+export const RejestracjaEmail = () => {
+
+  const dispatch = useDispatch();
+
+  const classes = useStyles()
+
+  const [imie, setImie] = useState('')
+  const [nazwisko, setNazwisko] = useState('')
+  const [email, setEmail] = useState('')
+  const [haslo, setHaslo] = useState('')
+
+  const [imieError, setImieError] = useState(false);
+  const [nazwiskoError, setNazwiskoError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [hasloError, setHasloError] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    setImieError(false);
+    setNazwiskoError(false)
+    setEmailError(false);
+    setHasloError(false);
+
+    if (imie === '') {
+      setImieError(true)
+    }
+
+    if (nazwisko === '') {
+      setNazwiskoError(true)
+    }
+
+    if (email === '') {
+      setEmailError(true)
+    }
+    if (haslo === '') {
+      setHasloError(true)
+    }
+  }
+
+  return (
+    <>
+      <ThemeProvider
+        theme={theme}>
+          <form
+            noValidate
+            autoComplete="off"
+            display="flex"
+            flexDirection="column"
+            onSubmit={handleSubmit}>
+
+            <TextField
+              className={classes.field}
+              id="outlined-basic"
+              label="Imię"
+              variant="outlined"
+              required
+              fullWidth
+              error={imieError}
+              onChange={(e) => setImie(e.target.value)}/>
+
+            <TextField
+              className={classes.field}
+              id="outlined-basic"
+              label="Nazwisko"
+              variant="outlined"
+              required fullWidth
+              error={nazwiskoError}
+              onChange={(e) => setNazwisko(e.target.value)}/>
+
+            <TextField
+              className={classes.field}
+              id="outlined-basic"
+              label="Email"
+              variant="outlined"
+              required
+              fullWidth
+              error={emailError}
+              onChange={(e) => setEmail(e.target.value)}/>
+
+            <TextField
+              className={classes.field}
+              id="outlined-basic"
+              label="Hasło"
+              variant="outlined"
+              required
+              fullWidth
+              error={hasloError}
+              onChange={(e) => setHaslo(e.target.value)} />
+
+            <Button
+              className={classes.field}
+              onClick={() => {
+                  if (imie && nazwisko && email && haslo !== '') {
```

Comment:
Nie jestem pewny czy to porównanie działa poprawnie, ale jeśli dodasz trim() w setach to można dać tu Boolean(imie) && i kolejne warunki.

Reply:
poprawione